Bio/SeqIO/TwoBitIO.py

```diff
@@ -0,0 +1,229 @@
+# as the DNA itself.
+#
+# The file begins with a 16-byte header containing the following fields:
+#
+# signature - the number 0x1A412743 in the architecture of the machine that
+#             created the file
+# version - zero for now. Readers should abort if they see a version number
+#           higher than 0
+# sequenceCount - the number of sequences in the file
+# reserved - always zero for now
+#
+# All fields are 32 bits unless noted. If the signature value is not as
+# given, the reader program should byte-swap the signature and check if the
+# swapped version matches. If so, all multiple-byte entities in the file
+# will have to be byte-swapped. This enables these binary files to be used
+# unchanged on different architectures.
+#
+# The header is followed by a file index, which contains one entry for each
+# sequence. Each index entry contains three fields:
+#
+# nameSize - a byte containing the length of the name field
+# name - the sequence name itself (in ASCII-compatible byte string), of
+#        variable length depending on nameSize
+# offset - the 32-bit offset of the sequence data relative to the start of
+#          the file, not aligned to any 4-byte padding boundary
+#
+# The index is followed by the sequence records, which contain nine fields:
+#
+# dnaSize - number of bases of DNA in the sequence
+# nBlockCount - the number of blocks of Ns in the file (representing unknown
+#               sequence)
+# nBlockStarts - an array of length nBlockCount of 32 bit integers
+#                indicating the (0-based) starting position of a block of Ns
+# nBlockSizes - an array of length nBlockCount of 32 bit integers indicating
+#               the length of a block of Ns
+# maskBlockCount - the number of masked (lower-case) blocks
+# maskBlockStarts - an array of length maskBlockCount of 32 bit integers
+#                   indicating the (0-based) starting position of a masked block
+# maskBlockSizes - an array of length maskBlockCount of 32 bit integers
+#                  indicating the length of a masked block
+# reserved - always zero for now
+# packedDna - the DNA packed to two bits per base, represented as so:
+#             T - 00, C - 01, A - 10, G - 11. The first base is in the most
+#             significant 2-bit byte; the last base is in the least significan
+#             2 bits. For example, the sequence TCAG is represented as 00011011.
+
+
+import numpy
+
+from Bio.Seq import Seq
+from Bio.SeqRecord import SeqRecord
+
+from .Interfaces import SequenceIterator
+from . import _twoBitIO
+
+
+class TwoBitSequenceData:
+    """Store the information needed to retrieve sequence data from a .2bit file.
+
+    Objects of this class store the file position at which the sequence data
+    start, the sequence length, and the start and end position of unknown (N)
+    and masked (lowercase) letters in the sequence.
+
+    Only two methods are provided: __len__ and __getitem__. The former will
```

Comment:
Is there value in defining ``__bytes__`` too?

Can you not define ``__repr__`` and perhaps even ``__str__`` as well?

Reply:
This class is not meant to be used by the end user, but only as the `data` argument when creating `Seq` object by `TwoBitIterator`. Therefore we want to create the absolute minimal class here. The `Seq` object will store the `data` argument as the `_data` attribute, but this attribute is not meant to be accessed by the user directly, only via the `Seq` object methods.

Reply:
Then give the class a leading underscore and add "(PRIVATE)" to the end of the class docstring?